### ISSUE TITLE
HTTP URI case reporting api version template parameter as violating.

### ISSUE THREAD

**Issue by cledoux** [googleapis/api-linter#897]
The camel-case-uris rule is returning false positive errors on the api version selector template variable `{$api_version}`.

Linter Rule: https://linter.aip.dev/122/camel-case-uris

Reproduction steps:

1.  Define a URI with the api version selector variable "{$api_version}". E.g. 
```
rpc GetBook(GetBookRequest) returns (Book) {
  option (google.api.http) = {
    get: "/{$api_version}/{name=publishers/*/books/*}"
  };
  option (google.api.method_signature) = "name";
}
```
2.  Run the linter.

Expected result:

No error.

Actual result:

Linter rule camel-case-uris error on  "/{$api_version}/{name=publishers/*/books/*}" due to the underscore in `{$api_version}`.
